### PR TITLE
feat: clear_value tactic

### PR DESCRIPTION
**Pull request by evanlohn**

### CODE REVIEW

File: Mathlib.lean [hunk 1]
```diff
@@ -61,8 +61,12 @@ import Mathlib.Tactic.Alias
 import Mathlib.Tactic.Basic
 import Mathlib.Tactic.Cache
 import Mathlib.Tactic.Cases
+<<<<<<< HEAD
 import Mathlib.Tactic.ClearExcept
 import Mathlib.Tactic.Clear_
+=======
+import Mathlib.Tactic.ClearValue
+>>>>>>> df532d4 (initial probably working version)
```
Comment (semorrison) [inline Mathlib.lean:69]: Please fix merge conflicts.

File: Mathlib/Tactic/ClearValue.lean [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+-/
+import Lean
+
+namespace Mathlib.Tactic.ClearValue
+open Lean Meta Elab Tactic Term
+
+syntax (name := clearValue) "clear_value" (ppSpace colGt ident)* : tactic
+
+
+def getNameAndLDecl [Monad m] [MonadError m] (ctx : LocalContext) (stx : Syntax) :
```
Comment (semorrison) [inline Mathlib/Tactic/ClearValue.lean:14]: Do we want dot notation here, so we can write `ctx.getNameAndLDecl`?

Comment (semorrison) [reply Mathlib/Tactic/ClearValue.lean:14]: I'd discourage introducing new abbreviations, i.e. don't abbreviate LocalDecl to LDecl unless this is already widespread.

File: test/ClearValue.lean [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  let c := a + b
+  clear_value c
+  have d := a + c
+  fail_if_success clear_value d
+
+  let e := 1
+  let f := 2
+  let g := d
+  let h := 4 + c
+  have _i := e+f + g +h
```
Comment (semorrison) [inline test/ClearValue.lean:13]:
```suggestion
  have _i := e + f + g + h
```

File: test/ClearValue.lean [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import Mathlib.Tactic.ClearValue
+
+example {a b: Nat} (h: a = b) : a = b := by
```
Comment (semorrison) [inline test/ClearValue.lean:3]:
```suggestion
example {a b : Nat} (h : a = b) : a = b := by 
```

File: test/ClearValue.lean [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import Mathlib.Tactic.ClearValue
+
+example {a b : Nat} (h : a = b) : a = b := by
```
Comment (digama0) [inline test/ClearValue.lean:3]: There should be a test including dependent types, where clearing the value would result in an ill-typed goal.

**Comment by urkud** [comment]
Should we close this PR now or there is some material (e.g., tests) that should be merged?

**Comment by semorrison** [comment]
I check the test suites, and I think the existing tests cover enough.

**Pull request closed without merging**